src/terms/ewald.jl

```diff
@@ -33,35 +33,52 @@ function energy_ewald(model::Model{T}; kwargs...) where {T}
     energy_ewald(model.lattice, charges, model.positions; kwargs...)
 end
 
+# To compute the electrostatics of the system, we use the Ewald splitting method due to the
+# slow convergence of the energy in ``1/r``.
+# It uses the the identity ``1/r ≡ erf(η·r)/r + erfc(η·r)/r``, where the first (smooth) part
+# of the energy term is computed in the reciprocal space and the second (singular) one in
+# the real-space.
+# `η` is an arbitrary parameter that enables to balance the computation of those to parts.
+# By default, we choose it to have a slight bias towards the reciprocal summation.
+function default_η(lattice::AbstractArray{T}) where {T}
+    any(iszero.(eachcol(lattice))) && return  # We won't compute anything
+    recip_lattice = compute_recip_lattice(lattice)
+    sqrt(sqrt(T(1.69) * norm(recip_lattice ./ 2T(π)) / norm(lattice))) / 2
+end
+
 # This could be merged with Pairwise, but its use of `atom_types` would slow down this
 # computationally intensive Ewald sums. So we leave it as it for now.
 """
-Compute the electrostatic interaction energy per unit cell between point charges in
-a uniform background of compensating charge to yield net neutrality.`lattice` should
-contain the lattice vectors as columns. `charges` and `positions` are the point charges and
-their positions (as an array of arrays) in fractional coordinates. If `forces` is not
-nothing, minus the derivatives of the energy with respect to `positions` is computed.
+Computes the local energy and forces on the atoms of the reference unit cell 0, for an
+infinite array of atoms at positions r_{iR} = positions[i] + R + ph_disp[i]*e^{iq·R} in
+a uniform background of compensating charge to yield net neutrality.
+`lattice` should contain the lattice vectors as columns. `charges` and `positions` are the
+point charges and their positions (as an array of arrays) in fractional coordinates. If
+`forces` is not nothing, minus the derivatives of the energy with respect to `positions` is
+computed.
+
+`q` is the phonon `q`-point (`Vec3`), and `ph_disp` a list of `Vec3` displacements to
+compute the Fourier transform of the force constant matrix.
 
 For now this function returns zero energy and force on non-3D systems. Use a pairwise
 potential term if you want to customise this treatment.
 """
-function energy_ewald(lattice::AbstractArray{T}, charges, positions;
-                      η=nothing, forces=nothing) where {T}
+function energy_ewald(lattice::AbstractArray{T}, charges, positions; η=default_η(lattice),
+                      forces=nothing, q=nothing, ph_disp=nothing) where {T}
```

Comment:
I'm not sure why we have this API of having the forces being modified... This is very old code from when we did not know julia very well. Can you change the API (here and in pairwise) to just accept a flag, and return the forces if the flag is on? (possibly in a separate PR if that's more convenient)